genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/grpc/GRpcAgentFileStreamServiceImpl.java

```diff
@@ -145,7 +146,7 @@ public synchronized void forceServerSync() {
         if (started.get()) {
             try {
                 this.jobDirectoryManifestCreatorService.invalidateCachedDirectoryManifest(jobDirectoryPath);
-                this.pushManifest();
+                this.taskScheduler.schedule(this::pushManifest, Instant.now());
```

Comment:
Perhaps if we want to force this it should be synchronous on the current thread instead of putting it on a scheduler?  I think even though you say `now` that's just a best effort to schedule it as soon as possible.

Reply:
I understand scheduling it `now` is just a best effort. 
I don't want the calling thread to get tangled in gRPC, but i could make this synchronous by waiting on the future returned by `schedule`.